rmc/utils/missense_badness.py

```diff
@@ -28,107 +26,11 @@
-    )
-    ht = ht.annotate(
-        ref=codon_map.get(ht.ref, "Unk"),
-        alt=codon_map.get(ht.alt, "Unk"),
-    )
-    # Remove any lines with "Unk" (unknown amino acids)
-    return ht.filter((ht.ref != "Unk") & (ht.alt != "Unk"))
-
-
-def get_oe_annotation(ht: hl.Table) -> hl.Table:
-    """
-    Annotate input Table with observed to expected missense (OE) ratio per transcript.
-
-    Use regional OE value if available, otherwise use transcript OE value.
-
-    .. note::
-        - Assumes input Table has `locus` and `trancript` annotations
-        - OE values are transcript specific
-        - Assumes merged RMC results HT exists
-        - Assumes merged RMC results HT is annotated per transcript section with:
-            - `section_oe`: Missense observed/expected ratio
-            - `interval`: Transcript section start position to end position
-
-    :param hl.Table ht: Input Table.
-    :return: Table with `oe` annotation.
-    """
-    overall_oe_ht = (
-        constraint_prep.ht().select_globals().select("total_obs", "total_exp")
-    )
-    group_ht = overall_oe_ht.group_by("transcript").aggregate(
-        obs=hl.agg.take(overall_oe_ht.total_obs, 1)[0],
-        exp=hl.agg.take(overall_oe_ht.total_exp, 1)[0],
-    )
-    # Recalculating transcript level OE ratio because previous OE ratio (`overall_oe`)
-    # is capped at 1 for regional missense constraint calculation purposes
-    group_ht = group_ht.annotate(transcript_oe=group_ht.obs / group_ht.exp)
-
-    # Read in LoF constraint HT to get OE ratio for five transcripts missing in v2 RMC results
-    # # 'ENST00000304270', 'ENST00000344415', 'ENST00000373521', 'ENST00000381708', 'ENST00000596936'
-    # All 5 of these transcripts have extremely low coverage in gnomAD
-    # Will keep for consistency with v2 LoF results but they look terrible
-    # NOTE: LoF HT is keyed by gene and transcript, but `_key_by_assert_sorted` doesn't work here for v2 version
-    # Throws this error: hail.utils.java.FatalError: IllegalArgumentException
-    lof_ht = constraint_ht.ht().select("oe_mis").key_by("transcript")
-    ht = ht.annotate(
-        gnomad_transcript_oe=lof_ht[ht.transcript].oe_mis,
-        rmc_transcript_oe=group_ht[ht.transcript].transcript_oe,
-    )
-    ht = ht.transmute(
-        transcript_oe=hl.coalesce(ht.rmc_transcript_oe, ht.gnomad_transcript_oe)
-    )
-
-    if not file_exists(rmc_results.path):
-        raise DataException("Merged RMC results table does not exist!")
-    rmc_ht = rmc_results.ht().key_by("interval")
-    ht = ht.annotate(
-        section_oe=rmc_ht.index(ht.locus, all_matches=True)
-        .filter(lambda x: x.transcript == ht.transcript)
-        .section_oe
-    )
-    ht = ht.annotate(
-        section_oe=hl.or_missing(
-            hl.len(ht.section_oe) > 0,
-            ht.section_oe[0],
-        ),
-    )
-    return ht.transmute(oe=hl.coalesce(ht.section_oe, ht.transcript_oe))
-
-
 def prepare_amino_acid_ht(
```

Comment:
we should update the call to this function to add freeze also (https://github.com/broadinstitute/regional_missense_constraint/blob/main/rmc/pipeline/calculate_missense_badness.py#L32)